nixos/modules/system/boot/systemd.nix

```diff
@@ -695,6 +697,20 @@ in
         minsize = "1M";
       };
     };
+
+    # make sure /run/next-system isn't garbage collected before the next boot
+    systemd.tmpfiles.rules = [ "L+ /nix/var/nix/gcroots/next-system - - - - /run/next-system" ];
+
+    # if /run/next-system exists, activate it before  rebooting
+    systemd.services.systemd-soft-reboot-activate-next-system = {
```

Comment:
What happened to doing this in `ExecStartPre` in `systemd-soft-reboot.service` with a `-` in the front of the command? That seemed quite elegant to me. With a new unit, I'm less sure that the unit ordering is right.

Reply:
I copied the ordering from prepare-kexec. I'm pretty confident this is right.


Using ExecStartPre= It would complain  about it not being able to connect to the journald.

```
May 07 21:11:02 utm (activate)[6377]: systemd-soft-reboot.service: Failed to connect stdout to the journal socket, ignoring: Connection refused
```

soft-reboot.target causes systemd-journald.service to exit before it is reached (has a Conflicts on it) to make sure logs are properly flushed before the switch is made. I think that's what causing this.